### DISCUSSION TITLE
[nvim][lspconfig ] elm-format complained and obstruct the editor on syntax error

### DISCUSSION THREAD

**Discussion by dtrckd** [elm-tooling/elm-language-server#961] [Q&A]
Hi there, 

I have set up elmls with nvim and [lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#elmls). Everything works fine, except one thing, when I format a file with the vim command `lua vim.lsp.buf.format()`, and that there is format errors in this file, I have a bunch of messages that are injected into the bottom bar log of vim, and that ask me to type <q> to return to normal mode. Furthermore, the number of these messages seems to accumulate and increase during a session, making it really annoying.

My nvim settings can be found here : https://github.com/dtrckd/config/blob/master/dotfiles/.config/nvim/lua/lsp_config.lua#L77

Here is the message 
```
**Request** Actions:                                                                                                                                                                                                     
Type number and <Enter> or click with the mouse (q or empty cancels): Running elm-format failed. Check the file for syntax errors.                                                                                   
Press ENTER or type command to continue                                                                                                                                                                              
Running elm-format failed. Check the file for syntax errors. 
Type number and <Enter> or click with the mouse (q or empty cancels):
```

If someone have an idea of what is going on, I will be happy to know more and solve this.

**Comment by hiddenroadss** [comment]
@dtrckd 
found a  solution here
https://elmlang.slack.com/archives/C0SR8T090/p1716574205492189?thread_ts=1716396451.370949&cid=C0SR8T090

**Reply by dtrckd** [reply]
Hi, I can't see the solution because I can't log in because slask tells me I don't have an account on this workspace :'(.

Could you copy the solution here?

**Reply by hiddenroadss** [reply]
https://github.com/joakin/nvim/blob/be72c11ff2d2c3ee6d6350f2221aabcca373adae/lua/plugins/lspconfig.lua#L148-L157
![image](https://github.com/user-attachments/assets/8f3a5e30-9b84-4ebb-a0f4-b0708c014800)

it is Elm Community slack channel, so you might like to join it anyway)

**Reply by kyu08** [reply]
@hiddenroadss @dtrckd 
Thanks! This helps me a lot!
